doc/contribute/modifying_contributions.rst

```diff
@@ -0,0 +1,67 @@
+.. _modifying_contributions:
+
+Modifying Contributions made by other developers
+************************************************
+
+Scenarios
+#########
+
+In general, Zephyr contributors and collaborators are encouraged to assist
+as reviewers in pull requests, so that patches may be approved and merged
+to Zephyr's main branch as part of the original pull requests. The authors
+of the pull requests are responsible for amending their original commits
+following the review process.
+
+There are occasions, however, when a contributor might need to modify patches
+included in pull requests that are submitted by other Zephyr contributors.
+For instance, this is the case when
+
+* a developer picks commits submitted by other contributors into their own
+  pull requests in order to
+
+  * integrate useful content which is part of a stale pull request, or
+  * get content merged to the project's main branch as part of a larger
+    patch
+
+* a developer pushes onto a branch or pull request opened by another
+  contributor in order to
+
+  * assist in revising pull requests and get the patches merged
+    to the project's main branch
+  * finalize review in stale pull requests
+
+
+Recommended policies
+####################
+
+A developer who intends to picking and modifying patches sent by
+another contributor should
+
+* clarify in their pull request the reason for doing that, instead
+  of assisisting in getting the patches merged in their original
+  pull request, and
+* invite the original author of the patches in their pull request review.
+
+A developer who intends to force-push onto a branch or pull request of
+another Zephyr contributor should clarify in the pull request the reason
+for pushing and for modifying the existing patches (e.g. stating that it
+is done for assisting in the pull request review when the pull
+request author is not able to do so).
+
+.. note::
+  Developers should try to limit the above practice to pull requests identified
+  as *stale*.
```

Comment:
The criteria and process for characterizing a PR as *stale* do not appear to be defined.

Also I sometimes do this on non-stale PRs when then original author agrees to let me, generally because of differential git skills and the complexity of getting the exact desired solution in place with a clean commit history.

Reply:
Agreed, this needs to be defined somewhere.

Reply:
@pabigot stale PRs and ISsues are defined here:
https://docs.zephyrproject.org/latest/development_process/dev_env_and_tools.html

I will add a link.

Reply:
> Also I sometimes do this on non-stale PRs when then original author agrees to let me, generally because of differential git skills and the complexity of getting the exact desired solution in place with a clean commit history.

Agreed. I believe, the way I wrote the sentence still allows you to continue doing this:
> should **try to limit** the above practice

But it is not encouraged as a long-term solution. Agreeing with original author is something we refer to later (for modifying content) but in theory it could also apply here, so I think we are sort of covered.